receiver/pulsarreceiver/factory.go

```diff
@@ -0,0 +1,155 @@
+		}
+	}
+}
+
+// WithLogsUnmarshalers adds LogsUnmarshalers.
+func WithLogsUnmarshalers(logsUnmarshalers ...LogsUnmarshaler) FactoryOption {
+	return func(factory *pulsarReceiverFactory) {
+		for _, unmarshaler := range logsUnmarshalers {
+			factory.logsUnmarshalers[unmarshaler.Encoding()] = unmarshaler
+		}
+	}
+}
+
+// NewFactory creates Pulsar receiver factory.
+func NewFactory(options ...FactoryOption) component.ReceiverFactory {
+
+	f := &pulsarReceiverFactory{
+		tracesUnmarshalers:  defaultTracesUnmarshalers(),
+		metricsUnmarshalers: defaultMetricsUnmarshalers(),
+		logsUnmarshalers:    defaultLogsUnmarshalers(),
+	}
+	for _, o := range options {
+		o(f)
+	}
+	return component.NewReceiverFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithTracesReceiverAndStabilityLevel(f.createTracesReceiver, stability),
+		component.WithMetricsReceiverAndStabilityLevel(f.createMetricsReceiver, stability),
+		component.WithLogsReceiverAndStabilityLevel(f.createLogsReceiver, stability),
+	)
+}
+
+type pulsarReceiverFactory struct {
+	tracesUnmarshalers  map[string]TracesUnmarshaler
+	metricsUnmarshalers map[string]MetricsUnmarshaler
+	logsUnmarshalers    map[string]LogsUnmarshaler
+}
+
+func (f *pulsarReceiverFactory) createTracesReceiver(
+	_ context.Context,
+	set component.ReceiverCreateSettings,
+	cfg config.Receiver,
+	nextConsumer consumer.Traces,
+) (component.TracesReceiver, error) {
+	c := *(cfg.(*Config))
+	if len(c.Topic) == 0 {
+		c.Topic = defaultTraceTopic
+	}
+	r, err := newTracesReceiver(c, set, f.tracesUnmarshalers, nextConsumer)
+	if err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+func (f *pulsarReceiverFactory) createMetricsReceiver(
+	_ context.Context,
+	set component.ReceiverCreateSettings,
+	cfg config.Receiver,
+	nextConsumer consumer.Metrics,
+) (component.MetricsReceiver, error) {
+	c := *(cfg.(*Config))
+	if len(c.Topic) == 0 {
+		c.Topic = defaultMeticsTopic
+	}
+	r, err := newMetricsReceiver(c, set, f.metricsUnmarshalers, nextConsumer)
+	if err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+func (f *pulsarReceiverFactory) createLogsReceiver(
+	_ context.Context,
+	set component.ReceiverCreateSettings,
+	cfg config.Receiver,
+	nextConsumer consumer.Logs,
+) (component.LogsReceiver, error) {
+	c := *(cfg.(*Config))
+	if len(c.Topic) == 0 {
+		c.Topic = defaultLogsTopic
+	}
+	r, err := newLogsReceiver(c, set, f.logsUnmarshalers, nextConsumer)
+	if err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+func createDefaultConfig() config.Receiver {
+	return &Config{
+		ReceiverSettings:           config.NewReceiverSettings(config.NewComponentID(typeStr)),
+		Topic:                      "",
+		Encoding:                   defaultEncoding,
+		ConsumerName:               defaultConsumerName,
+		Subscription:               defaultSubscription,
+		Endpoint:                   defaultServiceURL,
+		TLSAllowInsecureConnection: false,
+		Authentication:             Authentication{},
```

Comment:
Nit: no reason to specify zero values
```suggestion
		Encoding:                   defaultEncoding,
		ConsumerName:               defaultConsumerName,
		Subscription:               defaultSubscription,
		Endpoint:                   defaultServiceURL,
```